libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -209,6 +223,11 @@ - (SFSyncState*) syncDownWithTarget:(SFSyncDownTarget*)target options:(SFSyncOpt
 /** Resync
  */
 - (SFSyncState*) reSync:(NSNumber*)syncId updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
+    if (syncId.integerValue == self.syncIdRunning) {
```

Comment:
You are right. On the android side, everything is synchronous (smartstore / network) and dispatching the sync to a thread pool with a single worker guarantees that all syncs are done serially. But on iOS, networking is asynchronous, and so concurrent syncs are possible. There means bad things could happen if the developers are not careful. I feel we should limit concurrency (either completely like the current Android's implementation) or at least by soup.
For the resync fix, we could continue "living dangerously" and use a collection of "running sync ids" instead of a single "running sync id".